src/main/groovy/com/avast/gradle/dockercompose/ComposeExtension.groovy

```diff
@@ -1,144 +1,27 @@
-    boolean removeContainers = true
-    RemoveImages removeImages = RemoveImages.None
-    boolean removeVolumes = true
-    boolean removeOrphans = false
-    boolean forceRecreate = false
-
-    String executable = 'docker-compose'
-    Map<String, Object> environment = new HashMap<String, Object>(System.getenv())
-
-    String dockerExecutable = 'docker'
-
-    String dockerComposeWorkingDirectory = null
-    Duration dockerComposeStopTimeout = Duration.ofSeconds(10)
-
-    ComposeExtension(Project project, ComposeUp upTask, ComposeDown downTask) {
-        this.project = project
-        this.downTask = downTask
-        this.upTask = upTask
-        this.dockerExecutor = new DockerExecutor(this)
-        this.composeExecutor = new ComposeExecutor(this)
-
-        if (OperatingSystem.current().isMacOsX()) {
-            // Default installation is inaccessible from path, so set sensible
-            // defaults for this platform.
-            this.executable = '/usr/local/bin/docker-compose'
-            this.dockerExecutable = '/usr/local/bin/docker'
-        }
-    }
-
-    void isRequiredBy(Task task) {
-        task.dependsOn upTask
-        task.finalizedBy downTask
-        def ut = upTask // to access private field from closure
-        task.getTaskDependencies().getDependencies(task)
-                .findAll { Task.class.isAssignableFrom(it.class) && ((Task) it).name.toLowerCase().contains('classes') }
-                .each { ut.shouldRunAfter it }
-        if (task instanceof ProcessForkOptions) exposeAsEnvironment(task as ProcessForkOptions)
-        if (task instanceof JavaForkOptions) exposeAsSystemProperties(task as JavaForkOptions)
-    }
-
-    Map<String, ServiceInfo> getServicesInfos() {
-        upTask.servicesInfos
-    }
-
-    void exposeAsEnvironment(ProcessForkOptions task) {
-        servicesInfos.values().each { serviceInfo ->
-            serviceInfo.containerInfos.each { instanceName, si ->
-                if (instanceName.endsWith('_1')) {
-                    task.environment << createEnvironmentVariables(serviceInfo.name.toUpperCase(), si)
-                }
-                task.environment << createEnvironmentVariables(instanceName.toUpperCase(), si)
-            }
-        }
-    }
-
-    void exposeAsSystemProperties(JavaForkOptions task) {
-        servicesInfos.values().each { serviceInfo ->
-            serviceInfo.containerInfos.each { instanceName, si ->
-                if(instanceName.endsWith('_1')) {
-                    task.systemProperties << createSystemProperties(serviceInfo.name, si)
-                }
-                task.systemProperties << createSystemProperties(instanceName, si)
-            }
+import org.gradle.util.ConfigureUtil
+
+class ComposeExtension extends ComposeSettings {
+    ComposeExtension(Project project) {
+        super(project, '')
+    }
+
+    private HashMap<String, ComposeSettings> settings = [:]
+
+    def methodMissing(String name, def args) {
```

Comment:
haha you are starting to like dynamic languages, right? :)

but clever! :)

Reply:
Not so much :)